src/home/components/home-banner/HomeBanner.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router';
+import { getConfig, ensureConfig } from '@edx/frontend-platform';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  Button, Form, IconButton, useToggle,
+} from '@openedx/paragon';
+import { Search as SearchIcon } from '@openedx/paragon/icons';
+
+import { VideoModal } from '../../../generic';
+
+import messages from './messages';
+
+ensureConfig([
+  'HOMEPAGE_OVERLAY_HTML',
+  'SITE_NAME',
+  'ENABLE_COURSE_DISCOVERY',
+  'HOMEPAGE_PROMO_VIDEO_YOUTUBE_ID',
+], 'Home Banner');
+
+const HomeBanner = () => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+  const [searchValue, setSearchValue] = useState('');
+  const [isOpen, open, close] = useToggle(false);
+
+  const {
+    HOMEPAGE_OVERLAY_HTML,
+    SITE_NAME,
+    ENABLE_COURSE_DISCOVERY,
+    HOMEPAGE_PROMO_VIDEO_YOUTUBE_ID,
+  } = getConfig();
+
+  const handleSearch = () => navigate(`/courses?search_query=${searchValue}`);
+
+  const heading = HOMEPAGE_OVERLAY_HTML ? (
+    <div className="banner-title" dangerouslySetInnerHTML={{ __html: HOMEPAGE_OVERLAY_HTML }} />
```

Comment:
Let's disable the `react/no-danger `rule to get rid of the warning

```suggestion
    // eslint-disable-next-line react/no-danger
    <div className="banner-title" dangerouslySetInnerHTML={{ __html: HOMEPAGE_OVERLAY_HTML }} />
```

Reply:
fixed